lib/semantics/tools.h

```diff
@@ -265,6 +265,22 @@ template<typename T> std::optional<std::int64_t> GetIntValue(const T &x) {
   }
 }
 
+template<typename T> std::optional<bool> GetBoolValue(const T &x) {
+  if (const auto *expr{GetExpr(x)}) {
+    return evaluate::ToBool(*expr);
+  } else {
+    return std::nullopt;
+  }
+}
+
+template<typename T> std::optional<std::string> GetString(const T &x) {
```

Comment:
For consistency with the two preceding function this should be GetStringValue.

Reply:
Thanks for the comments! I will incoprate these.